### PR TITLE
some code cleanup 

### PR DESCRIPTION
**Pull request by lovettchris**
- separated model from UI code 
- added some needed error handling 
- added fps status.  
- Also made project file more compatible with various win sdk install versions, I could not build because my winsdk was newer at version 10.0.19041.0 instead of 10.0.18362.0.  Also no need to hard code Microsoft.NETCore.UniversalWindowsPlatform to 6.2.10, mine was 6.2.12. Removing the version lets VS use the latest which works.

Probably should do this for all samples.

### CODE REVIEW

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs [hunk 1]
```diff
@@ -165,141 +88,63 @@ private async Task InitCameraAsync()
...
-                }
-            }
-            return detections;
+            var x = (float)Math.Exp(val);
+            return x / (1.0f + x);
         }
 
         // draw bounding boxes on the output frame based on evaluation result
-        private async Task DrawBoxes(float[] results, VideoFrame frame)
+        private async Task DrawBoxes(List<Model.DetectionResult> detetions, VideoFrame frame)
```
Comment (martinb35) [inline Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs:138]: typo - can you correct this to "detections" or "detectionResults"?

**Review by martinb35 — APPROVED** [review]

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs (hunk 1 above)
Comment (lovettchris) [reply Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs]: woops, fixed.

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/YOLOv4ObjectDetection.csproj [hunk 2]
```diff
@@ -151,7 +153,6 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
-      <Version>6.2.10</Version>
```
Comment (martinb35) [inline Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/YOLOv4ObjectDetection.csproj:154]: removing this results in the warning here: https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/NU1604
"Project dependency 'PackageA' (<= 9.0.0) does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results."

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs [hunk 3]
```diff
@@ -331,46 +175,43 @@ private async Task DrawBoxes(float[] results, VideoFrame frame)
...
-            
-            return data.ToList<float>();
+            if (_timer == null)
+            {
+                // now start processing frames, no need to do more than 30 per second!
+                _timer = new DispatcherTimer()
+                {
+                    Interval = TimeSpan.FromMilliseconds(30)
+                };
+                _timer.Tick += OnTimerTick;
```
Comment (martinb35) [inline Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs:201]: are you missing a call to _timer.Start()? I built this locally and the tick event never got triggered.

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml [hunk 4]
```diff
@@ -9,10 +9,22 @@
...
+        <AppBar Grid.Row="0" IsOpen="True" IsSticky="True">
+            <StackPanel Orientation="Horizontal">
+                <AppBarButton Icon="Play" x:Name="button_go" Label="Go" Click="button_go_Click" ToolTipService.ToolTip="Start detecting objects in the video." />
+            </StackPanel>
+        </AppBar>
+        <Grid Grid.Row="1">
+            <CaptureElement x:Name="WebCam" Grid.Row="1" Stretch="Fill"  VerticalAlignment="Center" HorizontalAlignment="Center" Width="416" Height="416"/>
+            <Canvas x:Name="OverlayCanvas" VerticalAlignment="Center" HorizontalAlignment="Center" Width="416" Height="416"/>
+        </Grid>
+        <Grid Grid.Row="2" Background="#202040">
```
Comment (martinb35) [inline Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml:26]: I found this color pretty hard to read with black text. can you change the background color to something lighter, or the text to something closer to white?

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs (hunk 3 above)
Comment (lovettchris) [reply Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs:201]: woops, yeah, currently I'm having trouble with the checked in Yolo model, see https://github.com/microsoft/Windows-Machine-Learning/issues/404.  If you have a model that works I'd like to try it...

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml (hunk 4 above)
Comment (lovettchris) [reply Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml:26]: I fixed it by switching to a themed background color.

File: Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/YOLOv4ObjectDetection.csproj (hunk 2 above)
Comment (lovettchris) [reply Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/YOLOv4ObjectDetection.csproj]: Hmmm, you are right, I changed it back to 6.2.10 and mysteriously it is working for me now.  Weird.  Perhaps I was missing a solution restore before...

**Review by martinb35 — APPROVED** [review]

**Pull request merged**